File: 724 - Find Pivot Index.py
class Solution:
    def pivotIndex(self, nums: list[int]) -> int:
        res = -1
        ls = 0
        rs = sum(nums)
        for i in range(len(nums)):
            ls = ls + nums[i - 1] if i - 1 >= 0 else 0
            rs = rs - nums[i]
            print(ls, rs)
            if ls == rs:
                res = i
                break
        return res
            

s = Solution()
s.pivotIndex([1,7,3,6,5,6])

File: 724 - Find Pivot Index.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Arrays;

class Solution {
    
    public int pivotIndex(int[] nums) {
        int ls = 0, rs = Arrays.stream(nums).sum(), res = -1;
        for (int i = 0; i < nums.length; i++) {
            ls = ls + (i - 1 >= 0 ? nums[i - 1] : 0);
            rs = rs - nums[i];
//            System.out.println(ls + "   " + rs);
            if (ls == rs) {
//                return i;
                res = i;
                break;
            }
        }
        return res;
    }
    
}
